### PR TITLE
Feature export ary

### PR DESCRIPTION
**Pull request by bewakes**

### CODE REVIEW

File: ary/models.py [hunk 1]
```diff
@@ -279,3 +283,185 @@ def can_modify(self, user):
             (self.lead and self.lead.can_modify(user)) or
             (self.lead_group and self.lead_group.can_modify(user))
         )
+
+    def _load_schema(self, FieldClass, GroupClass=None):
```
Comment (bibekdahal) [inline ary/models.py]: GroupClass cannot be None based on the function definition.

A more suitable name may be `get_schema` or `create_schema` or `generate_schema`: this doesn't load schema into the object but rather generate a new schema.

Comment (bibekdahal) [reply ary/models.py]: `FieldClass` is also never used and should be removed. So the function definition can be something like:

```
def create_schema_for_group(self, GroupClass):
```

File: ary/models.py [hunk 2]
```diff
@@ -279,3 +283,185 @@ def can_modify(self, user):
...
+                    'options': {
+                        x['key']: x['title'] for x in field.get_options()
+                    }
+                }
+                for field in group.fields.all()
+            ] for group in groups
+        }
+        return schema
+
+    def _data_from_schema(self, schema, raw_data):
```
Comment (bibekdahal) [inline ary/models.py]: I think we should write all function names (especially functions that get or set values) as an imperative name, that starts with a verb. `get_data_from_schema` would be a better name.

Comment (bibekdahal) [reply ary/models.py]: Perhaps this method can be `static` or a separate utility function? It doesn't seem to use any data of `self`.

File: ary/models.py [hunk 3]
```diff
@@ -279,3 +283,185 @@ def can_modify(self, user):
...
+        return {
+            underscore_to_title(k):
+                v if not isinstance(v, list)
+                else [mapping[k](y) for y in v]
+            for k, v in methodology_raw.items()
+        }
+
+    def get_summary_json(self):
+        # functions to get exact value of an entry in summary group
+        def identity(x):
```
Comment (bibekdahal) [inline ary/models.py]: Duplicate `identity` function that is already defined in *utils/common.py*.

File: ary/models.py [hunk 4]
```diff
@@ -279,3 +283,185 @@ def can_modify(self, user):
...
+
+        # Formatting of underscored keywords, by default is upper case as given
+        # by default_format() function below
+        formatting = {
+            'priority_sector': lambda x: 'Most Unmet Needs Sectors',
+            'affected_location': lambda x: 'Settings Facing Most Humanitarian Issues'  # noqa
+        }
+
+        default_format = underscore_to_title   # function
+
```
Comment (bibekdahal) [inline ary/models.py]: Useless function definition as `underscote_to_title` is already defined.

File: export/views.py [hunk 5]
```diff
@@ -5,13 +5,17 @@
...
 
 from export.serializers import ExportSerializer
 from export.models import Export
 from project.models import Project
 
-from export.tasks_entries import export_entries
+from export.tasks import export_entries, export_assessment
+
+import logging
+logger = logging.getLogger('django')
```
Comment (bibekdahal) [inline export/views.py]: Is this used?

**Review by bibekdahal — CHANGES_REQUESTED** [review]

File: ary/models.py [hunk 6]
```diff
@@ -279,3 +283,185 @@ def can_modify(self, user):
...
+        def default_format(x):
```
Comment (bibekdahal) [inline ary/models.py]: Useless function definition as `underscote_to_title` is already defined.

**Comment by bibekdahal** [comment]
For future: always write commit message as an imperative sentence and start with a capital letter.

References:
https://github.com/the-deep/deeper/blob/develop/docs/git.md#commit-messages
https://chris.beams.io/posts/git-commit/

File: ary/models.py (hunk 1 above)
Comment (bewakes) [reply ary/models.py]: sahi ho dahal.

File: export/views.py (hunk 5 above)
Comment (bewakes) [reply export/views.py]: It was used, later I removed logs and forget this one. My bad.

File: ary/models.py (hunk 3 above)
Comment (bewakes) [reply ary/models.py]: Roger That

File: ary/models.py (hunk 6 above)
Comment (bewakes) [reply ary/models.py]: Copy That.

**Review by bibekdahal — APPROVED** [review]

**Pull request merged**